src/main.js

```diff
@@ -763,40 +732,44 @@ function handleMainWindowWebContentsCrashed() {
 // helper functions
 //
 
-function isTrustedURL(url) {
+function parseURL(url) {
   if (!url) {
-    return false;
+    return null;
   }
-
-  let parsedUrl = url;
+  let parsedURL = url;
   if (typeof url === 'string') {
```

Comment:
nit: if `url` isn't a string, we end up returning something that might not be an `URL`. Should we test for an `URL` and return that explicitly, but otherwise return `null`?

Reply:
Reworking this a bit ...

Reply:
So the update to this one returns the url if it is an instance of URL and exits. If not, it tries to convert it to a URL and returns that if successful, otherwise returns null.